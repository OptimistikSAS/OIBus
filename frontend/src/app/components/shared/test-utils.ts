import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
