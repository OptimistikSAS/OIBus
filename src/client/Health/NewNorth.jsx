import React, { useState } from 'react'
import { Button, Col, Form, Popover, PopoverBody, Row } from 'reactstrap'
import { nanoid } from 'nanoid'
import { ConfigContext } from '../context/configContext.jsx'
import validationNorth from '../North/Form/North.validation'
import { OIbSelect, OIbText } from '../components/OIbForm'

const NewNorth = () => {
  const { newConfig, dispatchNewConfig, apiList } = React.useContext(ConfigContext)
  const [name, setName] = React.useState('')
  const [api, setApi] = React.useState(apiList[0])

  const [openPopover, setOpenPopover] = useState(false)
  const applications = newConfig?.north?.applications ?? []

  const togglePopover = () => setOpenPopover(!openPopover)

  const addApplication = () => {
    if (!validationNorth.application.isValidName(name, applications?.map((application) => application.name) ?? [])) {
      togglePopover(false)
      dispatchNewConfig({
        type: 'addRow',
        name: 'north.applications',
        value: {
          id: nanoid(),
          name,
          api,
          enabled: false,
        },
      })
    }
  }

  const handleChange = (attributeName, value) => {
    switch (attributeName) {
      case 'name':
        setName(value)
        break
      case 'api':
      default:
        setApi(value)
        break
    }
  }

  return (
    <>
      <Button
        id="add-north"
        className="inline-button autosize oi-north-button"
        size="sm"
        outline
      >
        + North
      </Button>
      <Popover
        className="pop-container"
        trigger="legacy"
        placement="left"
        target="add-north"
        isOpen={openPopover}
        toggle={togglePopover}
      >
        <PopoverBody>
          <Form>
            <Row>
              <Col md="7">
                <OIbText
                  label="Application Name"
                  value={name}
                  name="name"
                  onChange={handleChange}
                  defaultValue=""
                  valid={() => validationNorth.application.isValidName(name, applications?.map((application) => application.name) ?? [])}
                />
              </Col>
              <Col md="5">
                <OIbSelect label="API" value={api} name="api" options={apiList} defaultValue={apiList[0]} onChange={handleChange} />
              </Col>
            </Row>
            <br />
            <Col md="2">
              <Button size="sm" id="icon-add" className="oi-add-button" color="primary" onClick={addApplication}>
                Add
              </Button>
            </Col>
          </Form>
        </PopoverBody>
      </Popover>
    </>

  )
}

export default NewNorth
