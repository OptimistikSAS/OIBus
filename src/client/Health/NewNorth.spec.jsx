/**
 * @jest-environment jsdom
 */
import React from 'react'
import ReactDOM from 'react-dom'
import { act, Simulate } from 'react-dom/test-utils'

import NewNorth from './NewNorth.jsx'

import newConfig from '../../../tests/testConfig'

let container

const dispatchNewConfig = jest.fn()

beforeEach(() => {
  React.useContext = jest.fn().mockReturnValue({ newConfig, dispatchNewConfig, apiList: ['API1', 'API2', 'API3'] })
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

describe('NewNorth', () => {
  test('display NewNorth page based on config', async () => {
    act(() => {
      ReactDOM.render(
        <NewNorth />,
        container,
      )
    })
    expect(container).toMatchSnapshot()

    React.useContext = jest.fn().mockReturnValue({
      newConfig: null,
      dispatchNewConfig,
      apiList: ['API1', 'API2', 'API3'],
    })

    act(() => {
      ReactDOM.render(
        <NewNorth />,
        container,
      )
    })
    expect(container).toMatchSnapshot()

    React.useContext = jest.fn().mockReturnValue({
      newConfig: {},
      dispatchNewConfig,
      apiList: ['API1', 'API2', 'API3'],
    })

    act(() => {
      ReactDOM.render(
        <NewNorth />,
        container,
      )
    })
    expect(container).toMatchSnapshot()

    React.useContext = jest.fn().mockReturnValue({
      newConfig: { north: {} },
      dispatchNewConfig,
      apiList: ['API1', 'API2', 'API3'],
    })

    act(() => {
      ReactDOM.render(
        <NewNorth />,
        container,
      )
    })
    expect(container).toMatchSnapshot()

    React.useContext = jest.fn().mockReturnValue({
      newConfig: { north: { applications: [] } },
      dispatchNewConfig,
      apiList: ['API1', 'API2', 'API3'],
    })

    act(() => {
      ReactDOM.render(
        <NewNorth />,
        container,
      )
    })
    expect(container).toMatchSnapshot()
  })

  // TODO: improve this test to check handleAddApplication
  test('check add pressed with "new_application" id', () => {
    act(() => {
      ReactDOM.render(
        <NewNorth />, container,
      )
    })
    Simulate.click(document.getElementById('add-north'))
    // Simulate.change(document.getElementById('name'), { target: { value: 'new_application' } })
    // Simulate.click(document.getElementById('icon-add'))
    expect(container).toMatchSnapshot()
  })
})
