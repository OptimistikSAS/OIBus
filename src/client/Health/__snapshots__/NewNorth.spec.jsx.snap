// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewNorth check add connector without name & with protocol 1`] = `
<div>
  <div
    class="icon-menu"
  >
    <div
      class="dropleft dropdown"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="icon-dropdown btn btn-secondary btn-sm"
        type="button"
      >
        <svg
          class="icon-dropdown-ellipsis"
          fill="currentColor"
          height="1em"
          id="dropdown-toggle"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 192 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
          />
        </svg>
      </button>
      <div
        aria-hidden="true"
        class="dropdown-menu"
        role="menu"
        tabindex="-1"
      >
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-add-south"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
          Add South
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-add-north"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
          Add North
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <div
            aria-hidden="true"
            id="icon-settings"
            role="button"
          >
            <svg
              class="icon-dropdown-item"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
              />
            </svg>
            Settings
          </div>
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-restart"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256.455 8c66.269.119 126.437 26.233 170.859 68.685l35.715-35.715C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.75c-30.864-28.899-70.801-44.907-113.23-45.273-92.398-.798-170.283 73.977-169.484 169.442C88.764 348.009 162.184 424 256 424c41.127 0 79.997-14.678 110.629-41.556 4.743-4.161 11.906-3.908 16.368.553l39.662 39.662c4.872 4.872 4.631 12.815-.482 17.433C378.202 479.813 319.926 504 256 504 119.034 504 8.001 392.967 8 256.002 7.999 119.193 119.646 7.755 256.455 8z"
            />
          </svg>
          Restart
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-shutdown"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M11 2.05V12h2V2.05c5.053.501 9 4.765 9 9.95 0 5.523-4.477 10-10 10S2 17.523 2 12c0-5.185 3.947-9.449 9-9.95z"
              />
            </g>
          </svg>
          Shutdown
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`NewNorth check add connector without protocol & with new name 1`] = `
<div>
  <div
    class="icon-menu"
  >
    <div
      class="dropleft dropdown"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="icon-dropdown btn btn-secondary btn-sm"
        type="button"
      >
        <svg
          class="icon-dropdown-ellipsis"
          fill="currentColor"
          height="1em"
          id="dropdown-toggle"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 192 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
          />
        </svg>
      </button>
      <div
        aria-hidden="true"
        class="dropdown-menu"
        role="menu"
        tabindex="-1"
      >
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-add-south"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
          Add South
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-add-north"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
          Add North
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <div
            aria-hidden="true"
            id="icon-settings"
            role="button"
          >
            <svg
              class="icon-dropdown-item"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
              />
            </svg>
            Settings
          </div>
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-restart"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256.455 8c66.269.119 126.437 26.233 170.859 68.685l35.715-35.715C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.75c-30.864-28.899-70.801-44.907-113.23-45.273-92.398-.798-170.283 73.977-169.484 169.442C88.764 348.009 162.184 424 256 424c41.127 0 79.997-14.678 110.629-41.556 4.743-4.161 11.906-3.908 16.368.553l39.662 39.662c4.872 4.872 4.631 12.815-.482 17.433C378.202 479.813 319.926 504 256 504 119.034 504 8.001 392.967 8 256.002 7.999 119.193 119.646 7.755 256.455 8z"
            />
          </svg>
          Restart
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-shutdown"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M11 2.05V12h2V2.05c5.053.501 9 4.765 9 9.95 0 5.523-4.477 10-10 10S2 17.523 2 12c0-5.185 3.947-9.449 9-9.95z"
              />
            </g>
          </svg>
          Shutdown
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`NewNorth check add connector without protocol & without new name 1`] = `
<div>
  <div
    class="icon-menu"
  >
    <div
      class="dropleft dropdown"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="icon-dropdown btn btn-secondary btn-sm"
        type="button"
      >
        <svg
          class="icon-dropdown-ellipsis"
          fill="currentColor"
          height="1em"
          id="dropdown-toggle"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 192 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
          />
        </svg>
      </button>
      <div
        aria-hidden="true"
        class="dropdown-menu"
        role="menu"
        tabindex="-1"
      >
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-add-south"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
          Add South
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-add-north"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
          Add North
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <div
            aria-hidden="true"
            id="icon-settings"
            role="button"
          >
            <svg
              class="icon-dropdown-item"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
              />
            </svg>
            Settings
          </div>
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-restart"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256.455 8c66.269.119 126.437 26.233 170.859 68.685l35.715-35.715C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.75c-30.864-28.899-70.801-44.907-113.23-45.273-92.398-.798-170.283 73.977-169.484 169.442C88.764 348.009 162.184 424 256 424c41.127 0 79.997-14.678 110.629-41.556 4.743-4.161 11.906-3.908 16.368.553l39.662 39.662c4.872 4.872 4.631 12.815-.482 17.433C378.202 479.813 319.926 504 256 504 119.034 504 8.001 392.967 8 256.002 7.999 119.193 119.646 7.755 256.455 8z"
            />
          </svg>
          Restart
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-shutdown"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M11 2.05V12h2V2.05c5.053.501 9 4.765 9 9.95 0 5.523-4.477 10-10 10S2 17.523 2 12c0-5.185 3.947-9.449 9-9.95z"
              />
            </g>
          </svg>
          Shutdown
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`NewNorth check change name with "new_application" id 1`] = `
<div>
  <div
    class="icon-menu"
  >
    <div
      class="dropleft dropdown"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        class="icon-dropdown btn btn-secondary btn-sm"
        type="button"
      >
        <svg
          class="icon-dropdown-ellipsis"
          fill="currentColor"
          height="1em"
          id="dropdown-toggle"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 192 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
          />
        </svg>
      </button>
      <div
        aria-hidden="true"
        class="dropdown-menu"
        role="menu"
        tabindex="-1"
      >
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-add-south"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
          Add South
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-add-north"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
          Add North
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <div
            aria-hidden="true"
            id="icon-settings"
            role="button"
          >
            <svg
              class="icon-dropdown-item"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
              />
            </svg>
            Settings
          </div>
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-restart"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256.455 8c66.269.119 126.437 26.233 170.859 68.685l35.715-35.715C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.75c-30.864-28.899-70.801-44.907-113.23-45.273-92.398-.798-170.283 73.977-169.484 169.442C88.764 348.009 162.184 424 256 424c41.127 0 79.997-14.678 110.629-41.556 4.743-4.161 11.906-3.908 16.368.553l39.662 39.662c4.872 4.872 4.631 12.815-.482 17.433C378.202 479.813 319.926 504 256 504 119.034 504 8.001 392.967 8 256.002 7.999 119.193 119.646 7.755 256.455 8z"
            />
          </svg>
          Restart
        </button>
        <button
          class="icon-dropdown-item dropdown-item"
          role="menuitem"
          tabindex="0"
          type="button"
        >
          <svg
            class="icon-dropdown-item"
            fill="currentColor"
            height="1em"
            id="icon-shutdown"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M11 2.05V12h2V2.05c5.053.501 9 4.765 9 9.95 0 5.523-4.477 10-10 10S2 17.523 2 12c0-5.185 3.947-9.449 9-9.95z"
              />
            </g>
          </svg>
          Shutdown
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`NewNorth display NewNorth page based on config 1`] = `<div />`;

exports[`NewNorth display NewNorth page based on config 2`] = `<div />`;

exports[`NewNorth display NewNorth page based on config 3`] = `<div />`;

exports[`NewNorth display NewNorth page based on config 4`] = `<div />`;

exports[`NewNorth display NewNorth page based on config 5`] = `<div />`;
